src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -381,37 +383,45 @@ void HwndTerminal::SendOutput(std::wstring_view data)
     {
         return;
     }
+    const auto lock = _terminal->LockForWriting();
     _terminal->Write(data);
 }
 
 HRESULT _stdcall CreateTerminal(HWND parentHwnd, _Out_ void** hwnd, _Out_ void** terminal)
 {
-    auto _terminal = std::make_unique<HwndTerminal>(parentHwnd);
-    RETURN_IF_FAILED(_terminal->Initialize());
+    auto publicTerminal = std::make_unique<HwndTerminal>(parentHwnd);
+
+    RETURN_IF_FAILED(publicTerminal->Initialize());
 
-    *hwnd = _terminal->GetHwnd();
-    *terminal = _terminal.release();
+    *hwnd = publicTerminal->GetHwnd();
+    *terminal = publicTerminal.release();
 
     return S_OK;
 }
 
 void _stdcall TerminalRegisterScrollCallback(void* terminal, void __stdcall callback(int, int, int))
+try
 {
-    auto publicTerminal = static_cast<HwndTerminal*>(terminal);
+    const auto publicTerminal = static_cast<HwndTerminal*>(terminal);
     publicTerminal->RegisterScrollCallback(callback);
 }
+CATCH_LOG()

```

Comment:
While making the necessary modifications, I noticed that all of the C ABI functions where missing try/catch clauses.